daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/ContractStateMachine.scala

```diff
@@ -229,16 +230,19 @@ class ContractStateMachine[Nid](mode: ContractKeyUniquenessMode) {
       }
     }
 
-    /**  Must be used to handle lookups iff [[com.daml.lf.transaction.ContractKeyUniquenessMode.byKeyOnly]] is false
+    /**  Must be used to handle lookups iff in [[com.daml.lf.transaction.ContractKeyUniquenessMode.Off]] mode
       */
     def handleLookup(lookup: Node.LookupByKey): Either[KeyInputError, State] = {
       // If the key has not yet been resolved, we use the resolution from the lookup node,
       // but this only makes sense if `activeState.keys` is updated by every node and not only by by-key nodes.
-      require(!mode.byKeyOnly, "This method can only be used if all key nodes are considered")
+      if (mode != ContractKeyUniquenessMode.Strict)
+        throw new UnsupportedOperationException(
+          "handleLookup can only be used if all key nodes are considered"
+        )
       visitLookup(lookup.templateId, lookup.key.key, lookup.result, lookup.result).left.map(Left(_))
     }
 
-    /** Must be used to handle lookups iff [[com.daml.lf.transaction.ContractKeyUniquenessMode.byKeyOnly]] is true
+    /** Must be used to handle lookups iff in [[com.daml.lf.transaction.ContractKeyUniquenessMode.Off]] mode
```

Comment:
Minor: Do you want to throw an `UnsupportedOperationException` in this method analogously to `handleLookup` if we're in the wrong mode?